odo/backends/url.py

```diff
@@ -133,7 +133,9 @@ def append_urlX_to_X(target, source, **kwargs):
         with open(target.path, 'wb') as fp:
             for chunk in iter(curry(r.read, chunk_size), ''):
                 fp.write(chunk)
-                return target
+                if not chunk:
```

Comment:
Of course, I see what you mean now.  Fixing it to `iter(curry(...), b'')` fixes the issue, and we can remove the test and break.  Thanks!